zerver/migrations/0298_fix_unread_messages_in_deactivated_streams.py

```diff
@@ -0,0 +1,23 @@
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    """
+    We're changing the stream deactivation process to make it mark all messages
+    in the stream as read. For things to be consistent with streams that have been
+    deactivated before this change, we need a migration to fix those old streams,
+    to have all messages marked as read.
+    """
+    dependencies = [
+        ('zerver', '0297_draft'),
+    ]
+
+    operations = [
+        migrations.RunSQL("""
+            UPDATE zerver_usermessage SET flags = flags | 1
+            FROM zerver_message
+            INNER JOIN zerver_stream ON zerver_stream.recipient_id = zerver_message.recipient_id
+            WHERE zerver_message.id = zerver_usermessage.message_id
+            AND zerver_stream.deactivated = true;
+        """),
+    ]
```

Comment:
I don't think that the migration will complete on a large production server; we'll likely need to do this in batches with 1 query per stream so that we can effectively use the relevant index?

Reply:
Hmm, I didn't think it would be that expensive. Split it going over streams one by one now.

Reply:
That was definitely was worth doing; with some quick testing of this code, individual deactivated streams processed thus far take between 0.001s and 72s; given that we have many thousands of deactivated streams in zulip.com, we certainly would not have been happy with the original implementation.